code/lib/cli/src/automigrate/fixes/upgrade-storybook-related-dependencies.ts

```diff
@@ -0,0 +1,160 @@
+import { cyan, underline, yellow } from 'chalk';
+import { getIncompatibleStorybookPackages } from '../../doctor/getIncompatibleStorybookPackages';
+import { valid, coerce } from 'semver';
+
+interface Options {
+  upgradable: { packageName: string; version: string }[];
+  problematicPackages: { packageName: string; version: string }[];
+}
+
+/**
+ * Is the user upgrading to the `latest` version of Storybook?
+ * Let's try to pull along some of the storybook related dependencies to `latest` as well!
+ *
+ * We communicate clearly that this migration is a helping hand, but not a complete solution.
+ * The user should still manually check for other dependencies that might be incompatible.
+ *
+ * see: https://github.com/storybookjs/storybook/issues/25731#issuecomment-1977346398
+ */
+export const upgradeStorybookRelatedDependencies = {
+  id: 'upgradeStorybookRelatedDependencies',
+  versionRange: ['*.*.*', '*.*.*'],
+  promptType: 'auto',
+  promptDefaultValue: false,
+
+  async check({ packageManager, storybookVersion }) {
+    const packageJson = await packageManager.readPackageJson();
+    const analyzed = await getIncompatibleStorybookPackages({
+      currentStorybookVersion: storybookVersion,
+      packageManager,
+      skipErrors: true,
+    });
+
+    const all = await packageManager.getAllDependencies();
+
+    const associated = Object.keys(all).filter((dep) => dep.includes('storybook'));
+    const detected = analyzed
+      .filter((m) => m.hasIncompatibleDependencies)
+      .map((m) => m.packageName);
+
+    const list = await Promise.all(
+      Array.from(new Set([...associated, ...detected])).map(async (packageName) => {
+        return {
+          packageName,
+          version: await packageManager.latestVersion(packageName).catch(() => null),
+        };
+      })
+    );
+
+    const data = list.reduce<Options>(
+      (acc, k) => {
+        if (k.version !== null) {
+          const { packageName, version } = k;
+          const upgradable = !(
+            !valid(k.version) ||
+            k.version === coerce(packageJson?.dependencies?.[k.packageName])?.toString() ||
+            k.version === coerce(packageJson?.devDependencies?.[k.packageName])?.toString() ||
+            k.version === coerce(packageJson?.peerDependencies?.[k.packageName])?.toString()
+          );
+
+          if (upgradable) {
+            acc.upgradable.push({ packageName, version });
+          } else {
+            acc.problematicPackages.push({ packageName, version });
+          }
+        }
+
+        return acc;
+      },
+      { upgradable: [], problematicPackages: [] }
+    );
+
+    if (data.upgradable.length > 0) {
+      return data;
+    }
+
+    return null;
+  },
+
+  prompt({ upgradable: list }) {
+    return dedent`
+      You're upgrading to the latest version of Storybook. We recommend upgrading the following packages:
+      ${list
+        .map(({ packageName, version }) => `- ${cyan(packageName)}@${cyan(version)}`)
```

Comment:
The output I got was:

```
│   You're upgrading to the latest version of Storybook. We       │
│   recommend upgrading the following packages:                   │
│   - eslint-plugin-storybook@0.8.0                               │
```

Can we make it so that it show something like:

```
│   You're upgrading to the latest version of Storybook. We       │
│   recommend upgrading the following packages:                   │
│   - eslint-plugin-storybook@0.6.0 (0.8.0 available)             │ 
```

Or

```
│   You're upgrading to the latest version of Storybook. We       │
│   recommend upgrading the following packages:                   │
│   - eslint-plugin-storybook: 0.6.0 => 0.8.0                     │ 
```

This way as a user I know the version I have installed vs the version which is available

Reply:
I can pull the version string from the `package.json`, but getting the exact installed version would add a lot of requests.

Why is it useful to know what your current version is?